GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -2616,12 +2620,7 @@ private void CompareWithCurrentBranchToolStripMenuItem_Click(object sender, Even
                 return;
             }
 
-            var headBranchName = Module.RevParse(headBranch);
-
-            if (headBranchName is not null)
-            {
-                UICommands.ShowFormDiff(baseCommit.ObjectId, headBranchName, baseCommit.Subject, headBranch);
-            }
+            UICommands.ShowFormDiff(baseCommit.ObjectId, CurrentCheckout, baseCommit.Subject, CurrentBranch.Value);
```

Comment:
Should be correct. But I could not test this with detached head because `CompareWithCurrentBranch` is disabled then.

Reply:
The difference in behavior is that return is early, with a popup. (That should not occur as you mention).